FeedStoreChallenge/ManagedCache.swift

```diff
@@ -0,0 +1,43 @@
+//
+//  ManagedCache.swift
+//  FeedStoreChallenge
+//
+//  Created by Joshua Bryson on 9/9/21.
+//  Copyright © 2021 Essential Developer. All rights reserved.
+//
+
+import CoreData
+
+@objc(ManagedCache)
+internal class ManagedCache: NSManagedObject {
+	@NSManaged internal var timestamp: Date
+	@NSManaged internal var feed: NSOrderedSet
+}
+
+extension ManagedCache {
+	static func deletePreviousCache(in context: NSManagedObjectContext) throws {
+		try find(in: context).map(context.delete).map(context.save)
+	}
+
+	static func newUniqueInstance(in context: NSManagedObjectContext) throws -> ManagedCache {
+		try deletePreviousCache(in: context)
+		return NSEntityDescription.insertNewObject(forEntityName: String(describing: ManagedCache.self), into: context) as! ManagedCache
+	}
+
+	class func find(in context: NSManagedObjectContext) throws -> ManagedCache? {
+		let request = ManagedCache.fetchRequest()
+		request.fetchLimit = 1
+		return try context.fetch(request).first as? ManagedCache
+	}
+
+	var localFeed: [LocalFeedImage] {
+		let feed = Array(feed) as! [ManagedFeedImage]
```

Comment:
Suggestion:
`NSOrderedSet` can be mapped to array using `compactMap`, it will help to avoid force casting
```suggestion
		let feed = feed.compactMap { $0 as? ManagedFeedImage }
```